test/functional/sessions.test.js

```diff
@@ -194,4 +195,26 @@ describe('Sessions', function () {
 
     generateTopologyTests(testSuites, testContext, testFilter);
   });
+
+  context('unacknowledged writes', () => {
+    it('should not include session for unacknowledged writes', {
+      metadata: { requires: { topology: 'single' } },
+      test: withMonitoredClient('insert', { clientOptions: { w: 0 } }, function (
+        client,
+        events,
+        done
+      ) {
+        client
+          .db('test')
+          .collection('foo')
+          .insertOne({ foo: 'bar' }, err => {
+            expect(err).to.not.exist;
```

Comment:
Hmm, I would expect the `err` to indeed exist. Looks like something is amiss here

Reply:
This is an implicit session, it is my understanding we're only throwing for explicit sessions. I added another test and fixed a throw v.s return in `applySession` bug.

Reply:
oops, you're right! While we're here suggest: 
```
expect(event).nested.property('command.writeConcern.w').to.equal(0);
expect(event).nested.property('command.lsid').to.not.exist;
```